tutorial/lesson_22_jit_performance.cpp

```diff
@@ -0,0 +1,174 @@
+    Var xi("xi"), yi("yi");
+    
+    // Let's focus on 8x8 subtiles, and then vectorize across X, and unroll across Y
+    output.tile(x, y, xi, yi, 8, 8).vectorize(xi).unroll(yi);
+
+    // For more advanced scheduling: 
+    //
+    // We can improve this even more by using the .in() directive (see Tutorial 19), 
+    // which allows us to interpose new Funcs in between input and output.
+    // 
+    // Here we can inject a block_transpose function to allow us to do 8 vectorized loads from the input.
+    // Func block_transpose("block_transpose"), block("block");
+    // block_transpose = input.in(output).compute_at(output, x).vectorize(x).unroll(y);
+    //
+    // And now lets reorder and vectorize in X across the block 
+    // block = block_transpose.in(output).reorder_storage(y, x).compute_at(output, x).vectorize(x).unroll(y);
+
+    return Pipeline(output);
+}
+
+int main(int argc, char **argv) {
+
+    
+    // Lets measure the performance of calling realize on a simple pipeline
+    {
+        Pipeline pipeline = make_pipeline();
+
+        // Create an 1024x1024 output buffer to hold the results 
+        Buffer<uint16_t> result(1024, 1024);
+
+        // Lets realize the pipeline (which implicitly JIT compiles, dynamically links and executes the code)
+        size_t count = 0;
+        double t = benchmark([&]() {
+            pipeline.realize(result);
+            ++count;
+        });
+
+        std::cout << "Execute Pipeline (realize): " << int(count / t) << " times/sec\n";
+    }
+
+    // This time let's explicitly invoke the JIT compile, before we realize and execute the pipeline
+    {
+        Pipeline pipeline = make_pipeline();
+        Buffer<uint16_t> result(1024, 1024);
+
+        // Let's explicitly JIT compile before we realize, which will cache the generated code
+        // associated with the Pipeline object. Now when we call realize, there's no code generation 
+        // overhead when we execute the pipeline.
+        pipeline.compile_jit();
+
+        size_t count = 0;
+        double t = benchmark([&]() {
+            pipeline.realize(result);
+            ++count;
+        });
+
+        std::cout << "Execute Pipeline (compile before realize): " << int(count / t) << " times/sec\n";
+    }
+
+    // Alternatively we could compile to a Callable object ...
+    {
+        Pipeline pipeline = make_pipeline();
+        Buffer<uint16_t> result(1024, 1024);
+
+        auto arguments = pipeline.infer_arguments();
+        const Target target = get_jit_target_from_environment();
+        Callable callable = pipeline.compile_to_callable(arguments, target);
+
+        // The Callable object acts as a convienient way of invoking the compiled code like
+        // a function call, using an argv like syntax for the argument list. It also caches 
+        // the JIT compiled code, so there's no code generation overhead when invoking the
+        // callable object and executing the pipeline.
+
```

Comment:
Suggestion: add a clause after this one that uses `compile_to_std_function()`; its benefit is mostly cleaner type checking/coercion, but the call overhead is also slightly lower than `Callable` so it may measure as (very slightly) faster (but even if not, having an example that demos it is worthwhile)